pkg/cfgfile/cfgfile.go

```diff
@@ -31,11 +31,76 @@ import (
 	"github.com/xeipuuv/gojsonschema"
 )
 
+// we're wrapping a function from the underlying io/ioutil package here so
+// that it's easier to mock when we define our *small* tests for this package...
+// first, define an interface for reading a file (by path)
+type reader interface {
+	ReadFile(s string) ([]byte, error)
+}
+
+// then define a type (struct) that will implement the ReadFile method
+type cfgReaderType struct{}
+
+// then define the implementation for that type; it simply calls the
+// underlying ioutil.Readfile() method and returns the result
+func (r *cfgReaderType) ReadFile(s string) ([]byte, error) {
+	return ioutil.ReadFile(s)
+}
+
+// then define a private variable that is of type reader (interface)
+var cfgReader reader
+
+// now, to make things even more testable, do the same for the process of
+// validating the schema; first, create an interface to wrap the underlying
+// method that will be used to validate the schema
+type schemaValidator interface {
+	ValidateSchema(schema, cfg string) []serror.SnapError
+}
+
+// then, define a type (struct) that will validate the underlying schema
+type schemaValidatorType struct{}
+
+// and define an implementation for that type that performs the schema validation
+func (r *schemaValidatorType) ValidateSchema(schema, cfg string) []serror.SnapError {
+	schemaLoader := gojsonschema.NewStringLoader(schema)
+	testDoc := gojsonschema.NewStringLoader(cfg)
+	result, err := gojsonschema.Validate(schemaLoader, testDoc)
+	var serrors []serror.SnapError
+	// Check for invalid json
+	if err != nil {
+		serrors = append(serrors, serror.New(err))
+		return serrors
+	}
+	// check if result passes validation
+	if result.Valid() {
+		return nil
+	}
+	for _, err := range result.Errors() {
+		serr := serror.New(errors.New("Validate schema error"))
+		serr.SetFields(map[string]interface{}{
+			"value":       err.Value(),
+			"context":     err.Context().String("::"),
+			"description": err.Description(),
+		})
+		serrors = append(serrors, serr)
+	}
+	return serrors
+}
+
+// then define a private variable that is of type reader (interface)
+var cfgValidator schemaValidator
+
+// and finally, initialize to set the instance of that variable
+func init() {
+	cfgReader = &cfgReaderType{}
+	cfgValidator = &schemaValidatorType{}
```

Comment:
personally, I'd prefer the former to the latter; modifying the `Read(...)` function to accept an additional input argument just so that we can run our `small` tests seems less clean than defining the instance variables (the instance variables are local after all and are not visible outside of the package)